app/src/androidTest/java/com/github/se/signify/ui/screens/profile/SettingsScreenTest.kt

```diff
@@ -95,93 +98,90 @@ class SettingsScreenTest {
   }
 
   @Test
-  fun testSaveButtonDisabledWhenTextFieldIsEmpty() {
-    // Ensure the TextField is empty
-    composeTestRule.onNodeWithText(userViewModel.userName.value).performTextClearance()
+  fun testPressingCancelDoesntUpdateName() {
+    val newName = "Updated Username"
 
-    // Click the Save button
-    composeTestRule.onNodeWithText("Save").performClick()
+    composeTestRule.onNodeWithTag("usernameTextField").performTextInput(newName)
+
+    composeTestRule.onNodeWithTag("usernameTextField").performImeAction()
+
+    composeTestRule.onNodeWithText("Cancel").performClick()
 
-    // Verify that updateUserName is not called since the input was empty
     verify(userRepository, never())
-        .updateUserName(Mockito.anyString(), Mockito.anyString(), anyOrNull(), anyOrNull())
+        .updateUserName(Mockito.anyString(), eq(newName), anyOrNull(), anyOrNull())
   }
 
   @Test
-  fun testCancelButtonClearsInput() {
-    // Simulate entering a new name
-    val newName = "Temporary Name"
-    composeTestRule.onNodeWithText(userViewModel.userName.value).performTextInput(newName)
-
-    // Simulate selecting a new profile picture
-    val newProfilePicturePath = "file:///path/to/new/profile/picture.jpg"
-    composeTestRule.runOnIdle {
-      userViewModel.updateProfilePictureUrl(newProfilePicturePath.toUri())
-    }
+  fun testDeleteProfilePictureShowsDialog() {
 
-    // Click the Cancel button
-    composeTestRule.onNodeWithText("Cancel").performClick()
+    composeTestRule.onNodeWithContentDescription("Delete Profile Picture").performClick()
 
-    // Verify that the TextField is cleared
-    composeTestRule.onNodeWithText("Temporary Name").assertDoesNotExist()
+    composeTestRule.onNodeWithText("Confirm Changes").assertIsDisplayed()
+  }
 
-    // Check that the original profile picture URL is restored
-    composeTestRule.runOnIdle {
-      val restoredProfilePictureUrl = userViewModel.profilePictureUrl.value
-      assert(restoredProfilePictureUrl != newProfilePicturePath) {
-        "Profile picture URL was not reset"
-      }
-    }
+  @Test
+  fun testDeleteProfilePictureConfirm() {
+
+    composeTestRule.onNodeWithContentDescription("Delete Profile Picture").performClick()
+
+    composeTestRule.onNodeWithText("Confirm").performClick()
+
+    verify(userRepository)
+        .updateProfilePictureUrl(Mockito.anyString(), eq(null), anyOrNull(), anyOrNull())
   }
 
   @Test
-  fun testEditProfilePictureIconClick() {
-    // Click the Edit Profile Picture icon
-    composeTestRule.onNodeWithContentDescription("Edit Profile Picture").performClick()
+  fun testDeleteProfilePictureCancel() {
+    composeTestRule.onNodeWithContentDescription("Delete Profile Picture").performClick()
+
+    composeTestRule.onNodeWithText("Cancel").performClick()
+
+    verify(userRepository, never())
+        .updateProfilePictureUrl(Mockito.anyString(), eq(null), anyOrNull(), anyOrNull())
```

Comment:
## Requested changes

When checking for `never()` you usually want to be as loose as possible. Does this really need to look for `eq(null)`?

Reply:
Fixed.